src/WebUI/Common/KubeCardWithTable.tsx

```diff
@@ -0,0 +1,224 @@
+/*
+    Copyright (c) Microsoft Corporation. All rights reserved.
+    Licensed under the MIT license.
+*/
+
+import { BaseComponent, css } from "@uifabric/utilities";
+import { CardContent, CustomCard } from "azure-devops-ui/Card";
+import { ITableRow } from "azure-devops-ui/Components/Table/Table.Props";
+import { CustomHeader, HeaderDescription, HeaderTitle, HeaderTitleArea, HeaderTitleRow, TitleSize } from "azure-devops-ui/Header";
+import { Link } from "azure-devops-ui/Link";
+import { IStatusProps, Status, StatusSize } from "azure-devops-ui/Status";
+import { ITableColumn, SimpleTableCell, Table, TwoLineTableCell } from "azure-devops-ui/Table";
+import { Tooltip } from "azure-devops-ui/TooltipEx";
+import { ArrayItemProvider } from "azure-devops-ui/Utilities/Provider";
+import * as React from "react";
+import { IVssComponentProperties } from "../Types";
+import "./KubeCardWithTable.scss";
+import { IResourceStatusProps, ResourceStatus } from "./ResourceStatus";
+import { format } from "azure-devops-ui/Core/Util/String";
+
+export interface ITableComponentProperties<T> extends IVssComponentProperties {
+    className?: string;
+    headingText?: string | JSX.Element;
+    headingDescription?: string;
+    hideHeaders?: boolean;
+    hideLines?: boolean;
+    items: T[];
+    columns: ITableColumn<T>[];
+    onItemActivated?: (event: React.SyntheticEvent<HTMLElement>, tableRow: ITableRow<any>, selectedItem: any) => void;
+    onItemSelected?: (event: React.SyntheticEvent<HTMLElement>, tableRow: ITableRow<any>, selectedItem: any) => void;
```

Comment:
does activated mean the row has been double-clicked ?

Reply:
two things, it could be double click or single, when we have option for singleClickActivation in table then this is single click otherwise double click.